### PR TITLE
Adding in IR Sensor to D11

### PR DESCRIPTION
**Pull request by johnisanerd**
This adds the IR Sensor to D11.  ** Needs a much more thorough testing! **

This needs a thorough testing and I'll point out a few highlights where we should be cautious.  In particular, we should run it with an Ultrasonic Sensor, temp sensor, and the servo.  I did test with the servo and it doesn't seem to interfere.

### CODE REVIEW

File: Software/Python/gopigo.py [hunk 1]
```diff
@@ -675,3 +677,52 @@ def check_version():
 			version=14
 
 	return version
+
```
Comment (johnisanerd) [inline Software/Python/gopigo.py]: @CleoQc this is the added function to gogpigo.py and it should be tested.  This works with our Keyes remote.

File: Software/Python/test_ir_rec.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Test IR Receiver
```
Comment (johnisanerd) [inline Software/Python/test_ir_rec.py]: @CleoQc this is a quick python test program I put in there to test the IR Receiver.  You should see numbers flash across the screen if you're running it, have an IR receiver in D11, and are pressing buttons on your remote control.

File: Firmware/fw_ver_17/firmware_update_17.sh [hunk 3]
```diff
@@ -0,0 +1,35 @@
+#! /bin/bash
```
Comment (johnisanerd) [inline Firmware/fw_ver_17/firmware_update_17.sh]: @RobertLucian and @CleoQc you should be able to run this to update your firmware on your GoPiGo2.

Comment (RobertLucian) [inline Firmware/fw_ver_17/firmware_update_17.sh]: Wow. I just saw the PR. Soo much code in here 😮 I got scared, initially.
Re-reading the comments in here, it seems like there are just 3 important files that changed.

I'll test them *thoroughly* and when this is a go, I'll add a review.

**Comment by RobertLucian** [comment]
@johnisanerd I've tested it and there are 3 things I've noticed:

1. The `test_ir_recv.py` script returns either `-1` or any other number (say `15`, or `14` or `11`). It does that alternatively. So you'd usually see something like `-1,-1,-1,-1,15,15,15,-1,15,15,15,15,15,15,-1,14`.

2. Buttons randomly stop working. After some time they either start working again, or a reboot is required. 

3. The ir stops working altogether (again sometimes), just as if you'd put it on hold. If feels like it's lagging.

-------------

Except the bug-fixing stuff, I also think, the `ir_read` function should return a string that's symbolic to what's written on the remote control. That also helps in the GoPiGo2-GoPiGo3 compatibility scenario.

As far as the rest of the functionalities, everything else worked just as expected. No problems on that side.

**Pull request closed without merging**